src/CoreWCF.Primitives/src/CoreWCF/Description/ServiceDescription.cs

```diff
@@ -322,6 +333,13 @@ public ServiceDescription(IEnumerable<IServiceBehavior> injectedBehaviors, IServ
             ServiceProvider = services;
             // Clone IServiceBehavior DI services implementing ICloneable to allow per service override.
             var behaviors = injectedBehaviors.ToList();
+
+            // TODO: Remove this check and the dynamic assembly load once we target net8
+            if (Environment.Version.Major >= 8 && s_isDynamicCodeSupported.Value)
```

Comment:
I don't think we need to worry about if dynamic code is supported. The reason we had that switch in the operation invoker is because the Expression implementation will be slower with AOT and that's important on a hot code path. This is at service startup so the performance isn't critical. We can accept the slower interpreted version of Expressions in that scenario. My concern with disabling this feature with aot is some behaviors would be used with the regular coreclr, but when using aot, they aren't applied so the service will behave differently.

Reply:
> I don't think we need to worry about if dynamic code is supported. The reason we had that switch in the operation invoker is because the Expression implementation will be slower with AOT and that's important on a hot code path. This is at service startup so the performance isn't critical. We can accept the slower interpreted version of Expressions in that scenario. 

Here i disabled the feature with AOT because my understanding is that we cannot dynamically load an assembly in AOT context but i may be wrong.

> My concern with disabling this feature with aot is some behaviors would be used with the regular coreclr, but when using aot, they aren't applied so the service will behave differently.

My idea is to bring the feature in CoreWCF 1.x targeting netstandard2.0 without AOT support so we can start to get usage. Then once we target net8.0, replace that code with AOT compatible code as we get not reflected access to the `IKeyedServiceProvider` type.

Reply:
@mconnew as discussed I dropped the Assembly.Load and the AOT flag.